### ISSUE TITLE
Inheritance problem. Model forward in loss functions

### ISSUE THREAD

**Issue by vsuarezpaniagua** [UKPLab/sentence-transformers#1841]
Most of the loss functions have a call to `self.model(sentence_feature)['sentence_embedding']`, but this raises a problem when training a SentenceTransformer inherited class and the `__call__` function is overridden.

Shouldn't it be better to call the `self.model.forward` function directly for these loss functions?

Thanks!
